paymentsheet/src/main/java/com/stripe/android/paymentelement/embedded/EmbeddedRowSelectionImmediateActionHandler.kt

```diff
@@ -0,0 +1,30 @@
+package com.stripe.android.paymentelement.embedded
+
+import com.stripe.android.core.injection.ViewModelScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.inject.Singleton
+
+internal typealias InternalRowSelectionCallback = () -> Unit
+
+@Singleton
+internal interface EmbeddedRowSelectionImmediateActionHandler {
+    fun handleImmediateRowSelectionCallback()
+}
+
+@Singleton
```

Comment:
I don't think we should mark this as singleton, as we're not relying on any behavior related to it being a singleton.